crates/stackable-webhook/src/tls.rs

```diff
@@ -171,32 +172,98 @@ impl TlsServer {
             // trait function on `IntoMakeServiceWithConnectInfo`
             let tower_service = router.call(remote_addr).await.unwrap();
 
-            tokio::spawn(async move {
-                // Wait for tls handshake to happen
-                let Ok(tls_stream) = tls_acceptor.accept(tcp_stream).await else {
-                    trace!(%remote_addr, "error during tls handshake connection");
-                    return;
-                };
-
-                // Hyper has its own `AsyncRead` and `AsyncWrite` traits and doesn't use tokio.
-                // `TokioIo` converts between them.
-                let tls_stream = TokioIo::new(tls_stream);
-
-                // Hyper also has its own `Service` trait and doesn't use tower. We can use
-                // `hyper::service::service_fn` to create a hyper `Service` that calls our app through
-                // `tower::Service::call`.
-                let hyper_service = service_fn(move |request: Request<Incoming>| {
-                    // We need to clone here, because oneshot consumes self
-                    tower_service.clone().oneshot(request)
-                });
-
-                if let Err(err) = hyper_util::server::conn::auto::Builder::new(TokioExecutor::new())
-                    .serve_connection_with_upgrades(tls_stream, hyper_service)
-                    .await
-                {
-                    warn!(%err, %remote_addr, "failed to serve connection");
+            let span = tracing::debug_span!("accept tcp connection");
+            tokio::spawn(
+                async move {
+                    let span = tracing::trace_span!(
+                        "accept tls connection",
+                        // otel.name = "accept tls connection",
+                        "otel.kind" = ?SpanKind::Server,
+                        "otel.status_code" = Empty,
+                        "otel.status_message" = Empty,
+                        "server.address" = Empty,
+                        "server.port" = Empty,
+                        "client.address" = Empty,
+                        "client.port" = Empty,
+                        "network.local.address" = Empty,
+                        "network.local.port" = Empty,
+                        "network.peer.address" = Empty,
+                        "network.peer.port" = Empty,
+                        // network.protocol.name = Empty, // at this layer, we don't know the application protocol
+                        // network.protocol.version = Empty, // doesn't make sense for tcp
+                        "network.transport" = "tcp",
+                        "network.type" = self.socket_addr.semantic_convention_network_type(),
+                    );
+
+                    if let Ok(local_addr) = tcp_stream.local_addr() {
```

Comment:
question: Does the local address ever change during runtime? If not, we could retrieve the address once instead of retrieving it for every incoming connection.

Reply:
The local address is whatever the connection happened on, so yes it _can_ change during runtime, but can't change per connection.

Further explanation: If the app listens on 0.0.0.0, then clients can hit any IP on any reachable interface.

Reply:
> Further explanation: If the app listens on 0.0.0.0, then clients can hit any IP on any reachable interface.

Ah, you are right. I didn't think about that possibility.

Reply:
Futhermore, the local_addr is only used here, and since it is modifying the span, it needs to be below it.

An alternative would be to deal with it directly on span creation, like:

```rust
                    let span = tracing::trace_span!(
                        "accept tls connection",
                        // otel.name = "accept tls connection",
                        "otel.kind" = ?SpanKind::Server,
                        "otel.status_code" = Empty,
                        "otel.status_message" = Empty,
                        "server.address" = tcp_stream.local_addr().unwrap_or(Empty).ip().to_string(),
                        "server.port" = tcp_stream.local_addr().unwrap_or(Empty).port() as u64,
                        "client.address" = Empty,
                        "client.port" = Empty,
                        "network.local.address" = tcp_stream.local_addr().unwrap_or(Empty).ip().to_string(),
                        "network.local.port" = tcp_stream.local_addr().unwrap_or(Empty).port() as u64,
                        "network.peer.address" = Empty,
                        "network.peer.port" = Empty,
                        "network.transport" = "tcp",
                        "network.type" = self.socket_addr.semantic_convention_network_type(),
                    );
```

But that seems messy.

Reply:
^ Actually that doesn't work, because the unwrap_or wants a SocketAddr

Reply:
Yeah, that is messy indeed. Let's leave it as is.